github/git_commits.go

```diff
@@ -87,6 +90,29 @@ func (s *GitService) GetCommit(ctx context.Context, owner string, repo string, s
 	return c, resp, nil
 }
 
+// ListCommits fetches latest commits
+func (s *GitService) ListCommits(ctx context.Context, owner string, repo string, sha string) (*CommitList, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/git/commits", owner, repo)
+	req, err := s.client.NewRequest("GET", u, nil)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	q := req.URL.Query()
+	q.Add("sha", sha)
+	req.URL.RawQuery = q.Encode()
+
+	commits := new(CommitList)
+	resp, err := s.client.Do(ctx, req, commits)
+
```

Comment:
Please delete this line to closely bind the `if err != nil` test to the statement it is checking as per idiomatic Go.